tests/neo4j/txrun.py

```diff
@@ -118,18 +122,143 @@ def test_should_fail_run_in_a_rollbacked_tx(self):
             tx.run("RETURN 42").next()
 
     def test_should_fail_to_run_query_for_invalid_bookmark(self):
-        session = self._driver.session("w")
-        tx1 = session.beginTransaction()
+        self._session = self._driver.session("w")
+        tx1 = self._session.beginTransaction()
         result = tx1.run('CREATE ()')
         result.consume()
         tx1.commit()
-        lastBookmarks = session.lastBookmarks()
-        assert len(lastBookmarks) == 1
-        lastBookmark = lastBookmarks[0]
-        invalidBookmark = lastBookmark[:-1] + "-"
-        session.close()
-        session = self._driver.session("w", [invalidBookmark])
+        last_bookmarks = self._session.lastBookmarks()
+        assert len(last_bookmarks) == 1
+        last_bookmark = last_bookmarks[0]
+        invalid_bookmark = last_bookmark[:-1] + "-"
+        self._session.close()
+        self._session = self._driver.session("w", [invalid_bookmark])
 
         with self.assertRaises(types.responses.DriverError):
-            tx2 = session.beginTransaction()
+            tx2 = self._session.beginTransaction()
             tx2.run("CREATE ()").consume()
+
+    def test_tx_configuration(self):
+        self._session = self._driver.session("r")
+        tx = self._session.beginTransaction(txMeta={"foo": "bar"}, timeout=3)
+        result = tx.run("UNWIND [1,2,3,4] AS x RETURN x")
+        values = []
+        if self.driver_supports_features(types.Feature.TMP_RESULT_KEYS):
+            self.assertEqual(result.keys(), ["x"])
+        for record in result:
+            values.append(record.values[0])
+        tx.commit()
+        self.assertEqual(values, list(map(types.CypherInt, range(1, 5))))
+
+    def test_consume_after_commit(self):
+        self._session = self._driver.session("w", fetchSize=2)
+        tx = self._session.beginTransaction()
+        result = tx.run("UNWIND [1,2,3,4] AS x RETURN x")
+        if self.driver_supports_features(types.Feature.TMP_RESULT_KEYS):
+            self.assertEqual(result.keys(), ["x"])
+        values = []
+        for _ in range(2):
+            record = result.next()
+            self.assertIsInstance(record, types.Record)
+            values.append(record.values[0])
+        tx.commit()
+
+        # TODO: what should happen here?
```

Comment:
Which piece of code does this todo refer?

Reply:
There should be an assertion, but all official drivers behave differently here. So before we add something here, we need to decide what that something should be. https://trello.com/c/F1JAhHbz/81-unmanaged-tx-result-scope